File: app/src/main/java/com/esoxjem/movieguide/details/MovieDetailsInteractor.java
package com.esoxjem.movieguide.details;

import com.esoxjem.movieguide.Review;
import com.esoxjem.movieguide.Video;

import java.util.List;

import io.reactivex.Observable;

/**
 * @author arun
 */
public interface MovieDetailsInteractor
{
    Observable<List<Video>> getTrailers(String id);
    Observable<List<Review>> getReviews(String id);
}

File: app/src/main/java/com/esoxjem/movieguide/movies/details/MovieDetailsInteractor.kt
package com.esoxjem.movieguide.movies.details

import com.esoxjem.movieguide.movies.entities.Review
import com.esoxjem.movieguide.movies.entities.Video
import com.esoxjem.movieguide.network.TmdbWebService
import io.reactivex.Observable

/**
 * @author arunsasidharan
 */
class MovieDetailsInteractor(private val tmdbWebService: TmdbWebService) : MovieDetailsContract.Interactor {

    override fun getTrailers(id: String): Observable<List<Video>> {
        return tmdbWebService.trailers(id).map { it.videos }
    }

    override fun getReviews(id: String): Observable<List<Review>> {
        return tmdbWebService.reviews(id).map { it.reviews }
    }

}
